libraries/chain/smt_evaluator.cpp

```diff
@@ -132,26 +132,33 @@ static void verify_accounts( database& db, const flat_map< unit_target_type, uin
    }
 }
 
-struct smt_setup_evaluator_visitor
+struct smt_generation_policy_validator
```

Comment:
For consistency in naming between validate / evaluate, let's rename this `smt_setup_ico_tier_evaluator_visitor`

Reply:
This is already consistent because the evaluator aprt is called `smt_generation_policy_visitor`. I chose to name it with way because the validate/evaluate is not particular to `smt_setup_ico_tier_operation`, it is for any `smt_generation_policy`.